### PR TITLE
Change WarningLevel to 9999

### LINKED ISSUE #46462: Enable warning wave in Roslyn build
Once we are building with a sufficiently new compiler, we should update the Roslyn solutions and projects to enable warning waves, by changing WarningLevel from 4 to 9999.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Closes #46462

It turns out that we are just so great that we never hit any of our wave 5 warnings. (I did test adding some code that causes a wave 5 warning to verify that this config change does what we expect.)

Tagging @jmarolf for review.

### CODE REVIEW

**Review by sharwell — APPROVED** [review]

File: eng/targets/Settings.props [hunk 1]
```diff
@@ -195,7 +195,7 @@
     <When Condition="'$(Language)' == 'C#'">
       <PropertyGroup>
         <LangVersion>preview</LangVersion>
-        <WarningLevel>4</WarningLevel>
+        <WarningLevel>9999</WarningLevel>
```
Comment (alrz) [inline eng/targets/Settings.props:198]: Wasn't this supposed to be [`<AnalysisLevel>latest</AnalysisLevel>`](
 https://devblogs.microsoft.com/dotnet/automatically-find-latent-bugs-in-your-code-with-net-5/)?

Comment (Youssef1313) [reply eng/targets/Settings.props:198]: @alrz From what I understand, AnalysisLevel is related to the shipped analyzers (CAxxxx) in .NET SDK, not the compiler diagnostics. See https://docs.microsoft.com/dotnet/fundamentals/code-analysis/overview#latest-updates

Comment (RikkiGibson) [reply eng/targets/Settings.props:198]: perhaps we should adopt AnalysisLevel instead anyway..?

Comment (alrz) [reply eng/targets/Settings.props:198]: it is about analyzers but from the article it does affect warning level as well.

Comment (Youssef1313) [reply eng/targets/Settings.props:198]: I couldn't see where in the article it mentions warning waves, but if it does affect it, then definitely go with it. If not, then probably add both.

**Comment by RikkiGibson** [comment]
We'll wait on this until we can decide whether to adopt AnalysisLevel instead with @jmarolf's help. I was concerned that maybe this would opt us in to the analyzers baked-in to the .NET SDK and be either redundant or undesired (or needs to be controlled via the "-runAnalyzers" flag in our build scripts, etc.)

File: eng/targets/Settings.props (hunk 1 above)
Comment (jmarolf) [reply eng/targets/Settings.props:198]: I recommend setting this to `<AnalysisLevel>preview</AnalysisLevel>` only if we also want the additional analyzers to be brought along for the ride. As roslyn has rather complex analyzer consumption requirements (as we both build and consume the same analyzers in our build) I recommend `<WarningLevel>9999</WarningLevel>`

**Review by jmarolf — APPROVED** [review]

**Pull request merged**